BraveRewardsTests/LedgerDatabaseTests.mm

```diff
@@ -144,6 +147,638 @@ - (void)testNonZeroExcludedCount
+  filter.excluded = BATExcludeFilterFilterAllExceptExcluded;
+  
+  result = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:0 filter:filter];
+  XCTAssertEqual(result.count, 4);
+  
+  // Other excludes
+  filter.excluded = BATExcludeFilterFilterDefault;
+  
+  result = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:0 filter:filter];
+  XCTAssertEqual(result.count, 2);
+  
+  filter.excluded = BATExcludeFilterFilterIncluded;
+  
+  result = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:0 filter:filter];
+  XCTAssertEqual(result.count, 1);
+}
+
+- (void)testPublishersWithFiltersPercent
+{
+  const auto percent = 30;
+  
+  [self createBATPublisherInfo:@"1" reconcileStamp:20 percent:percent createActivityInfo:true];
+  [self createBATPublisherInfo:@"2" reconcileStamp:30 percent:80 createActivityInfo:true];
+  [self createBATPublisherInfo:@"3" reconcileStamp:40 percent:0 createActivityInfo:true];
+  [self createBATPublisherInfo:@"4" reconcileStamp:40 percent:5 createActivityInfo:true];
+  
+  const auto filter = [[BATActivityInfoFilter alloc] init];
+  filter.percent = percent;
+  
+  const auto result = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:0 filter:filter];
+  XCTAssertEqual(result.count, 2);
+  
+  filter.percent = 90;
+  
+  const auto emptyResult = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:0 filter:filter];
+  XCTAssertEqual(emptyResult.count, 0);
+}
+
+- (void)testPublishersWithFiltersVisits
+{
+  const auto visits = 10;
+  
+  // Same `duration` as reconcile stamp
+  [self createBATPublisherInfo:@"1" reconcileStamp:10 percent:1 createActivityInfo:YES duration:100
+                        visits:visits excluded:BATPublisherExcludeDefault];
+  
+  // `duration` higher than reconcile stamp
+  [self createBATPublisherInfo:@"2" reconcileStamp:30 percent:1 createActivityInfo:YES duration:350
+                        visits:19 excluded:BATPublisherExcludeDefault];
+  
+  [self createBATPublisherInfo:@"3" reconcileStamp:30 percent:1 createActivityInfo:YES duration:29
+                        visits:5 excluded:BATPublisherExcludeDefault];
+  
+  const auto filter = [[BATActivityInfoFilter alloc] init];
+  filter.minVisits = visits;
+  
+  const auto result = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:0 filter:filter];
+  XCTAssertEqual(result.count, 2);
+  
+  filter.minVisits = 100;
+  
+  const auto emptyResult = [BATLedgerDatabase publishersWithActivityFromOffset:0 limit:0 filter:filter];
+  XCTAssertEqual(emptyResult.count, 0);
+}
+
+#pragma mark - Media Publisher Info
+
+- (void)testInsertOrUpdateMediaPublisherInfoWithMediaKey
+{
+  [self createBATPublisherInfo:@"1" reconcileStamp:40 percent:0 createActivityInfo:true];
+  
+  XCTAssert([self countMustBeEqualTo:0 forEntityName:@"MediaPublisherInfo"]);
+  
+  [self waitForFailedSaveAttempt:^{
+  [BATLedgerDatabase insertOrUpdateMediaPublisherInfoWithMediaKey:@"" publisherID:@"1"];
```

Comment:
Minor/Lint: Spacing before this call. I assume the wait and fail was just added after :)